File: 600/605 - Can Place Flowers.java
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        boolean pv = true;
        int len = flowerbed.length;

        for (int i = 0; n > 0 && i < len - 1; ++i) {
            if (pv && flowerbed[i] == 0 && flowerbed[i + 1] == 0) {
                pv = false;
                --n;
            } else {
                pv = flowerbed[i] == 0;
            }
        }
        
        if (pv && n > 0 && flowerbed[len - 1] == 0) {
            --n;
        }
        
        return n == 0;
    }
}

File: csharp/605.can-place-flowers.cs
/*
 * @lc app=leetcode id=605 lang=csharp
 *
 * [605] Can Place Flowers
 */

using System.Linq;

// @lc code=start
public class Solution {
    public bool CanPlaceFlowers(int[] flowerbed, int n) {
        int len = 1;
        foreach (int c in flowerbed) {
            if (c == 0) {
                ++len;
                continue;
            }
            if (len > 2) {
                n -= (len - 1) / 2;
                if (n <= 0) {
                    return true;
                }
            }
            len = 0;
        }
        ++len;
        if (len > 2) {
            n -= (len - 1) / 2;
        }
        return n <= 0;
    }
}
// @lc code=end

